src/madwifi.c

```diff
@@ -821,15 +821,16 @@ check_devname (const char *dev)
 
 	if (dev[0] == '.')
 		return 0;
-	
+
 	ssnprintf (buf, sizeof (buf), "/sys/class/net/%s/device/driver", dev);
-	buf[sizeof (buf) - 1] = 0;
+	buf[sizeof (buf) - 1] = '\0';
 
-	memset (buf2, 0, sizeof (buf2));
```

Comment:
I don't understand why the old code is broken: `buf2` is initialized to zero, then `readlink()` will write at most `sizeof(buf2)-1` bytes, so the trailing null byte should not be overwritten. It's not the best style, but not broken, right?